src/System.Private.CoreLib/shared/System/Collections/HashHelpers.cs

```diff
@@ -86,5 +88,33 @@ public static int ExpandPrime(int oldSize)
 
             return GetPrime(newSize);
         }
+
+        // Takes an index and distributes it in the range [0, range)
+        // e.g. 0 to range exclusive.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static uint FastRange(uint index, uint range)
+        {
+            // Sse42.IsSupported check is baked into the R2R image so while the value returned
+            // from this method will be different on both branches, it will remain stable during
+            // a process' lifetime even if it is recompiled by tiered compilation.
+            if (Sse42.IsSupported)
+            {
+                // Using fastrange from Daniel Lemire https://lemire.me/blog/2016/06/27/a-fast-alternative-to-the-modulo-reduction/
+                //
+                // While fastrange is "fair"; it expects an evenly distributed hashCode
+                // from [0,2^32) to produce a distibuted range. As many hashCodes in .NET
+                // are generated with the high entropy in the LSB we need to redistribute
+                // the hash; for this we use Crc32. These two together allow us to skip the
+                // more costly idiv instruction.
+
+                const uint CrcReflected = 0xEDB88320;
+                return (uint)((Sse42.Crc32(CrcReflected, index) * (ulong)range) >> 32);
```

Comment:
Arm has Crc that's not currently exposed via intrinsics? Also its ` rbit` instruction would probably work instead to reverse the bits.

Reply:
As I wrote in the original PR, I am not quite sure why we are using the reflected polynomial of CRC-32 (`CrcReflected`) as the initial hash value for CRC-32C calculation implemented by the intrinsic.  Those are two different CRC methods, and the initial value usually defaults to `-1`.

Reply:
To reverse the output bit order; though it didn't seem hugely significant in testing.

Reply:
> To reverse the output bit order

How does that reverse better than `-1`?  Why are we using this particular constant taken from a different CRC method as the initial value?

Reply:
Ahh, I understand now. Changed to -1